src/kubernetes.cc

```diff
@@ -961,6 +961,18 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidateConfiguration() const {
+  try {
+    json::value raw_node = QueryMaster(
+        std::string(kKubernetesEndpointPath) + "/nodes/" + CurrentNode());
```

Comment:
`CurrentNode()` may be empty if, e.g., the API token is invalid (so you'd get 403s). Should we check that as well?

Reply:
Great point, done.